subsys/bluetooth/controller/Kconfig

```diff
@@ -431,6 +431,7 @@ config BT_CTLR_DATA_LENGTH
 config BT_CTLR_DATA_LENGTH_MAX
 	int "Maximum data length supported"
 	depends on BT_CTLR_DATA_LENGTH
+	default 37 if BT_MESH_GATT_SERVER
```

Comment:
@cvinayak has there been some policy wrt controller Kconfig options having dependencies on host Kconfig options? In most cases I think it should be discouraged, since the ability to do so is the special case where the host is part of the same Zephyr build (i.e. runs on the same core), thereby putting split host/controller architectures in a disadvantaged situation. That said, I do understand the argument that if we can be nice to unified builds (with both host and controller) without harming anything else, then we should do that.

I took a quick look at the rest of the Kconfig file, and I didn't manage to find any other places with references to host-specific Kconfig options.

Reply:
@cvinayak ping

Reply:
> has there been some policy wrt controller Kconfig options having dependencies on host Kconfig options?

No written policy. Controller code tries to use explicit Controller-only Kconfig options and unconditional defaults, except when Bluetooth Core common Kconfig (ex. BT_OBSERVER, BT_CENTRAL, BT_HCI_RAW etc.) can be used or the values can be inherited  (like BT_EXT_ADV_MAX_ADV_SET or BT_PER_ADV or BT_PER_ADV_SYNC).

> In most cases I think it should be discouraged

Yes, I discourage conditional Controller Kconfig default values inherited from outside the Bluetooth Core Specification recommendations. These are redundant when Host-only or Controller-only binary is built.

As Zephyr Host can interface with non-Zephyr native Controller, I suggest the override value be set using specific (overlay) conf file when using Zephyr Controller (include `CONFIG_BT_LL_SW_SPLIT=y`).

Reply:
@alxelax given the comments above, it would be great if you could find a way to do this without adding Host or Mesh dependencies in the controller